jcl/src/java.base/share/classes/java/lang/String.java

```diff
@@ -2563,6 +2563,26 @@ public String stripTrailing() {
 		
 		return (result == null) ? this : result;
 	}
+
+	/**
+	 * Determine if the string contains only white space characters. 
+	 * 
+	 * @return true if the string is empty or contains only white space
+	 * characters, otherwise false.
+	 * 
+	 * @since 11
+	 */
+	public boolean isBlank() {
+		int index;
+
```

Comment:
The first  thing `indexOfNonWhitespace` does is check the length of the array (which is basically the equivalent of `if (count == 0)`. I think we can hoist that check out and skip the rest if its zero. This has benefits when running interpreted, not sure how much this matters to compiled code. 
  
@fjeremic  thoughts?

Reply:
I don't think there is benefit to duplicating the check. It makes the method larger and less susceptible to inlining at basically no benefit. I don't think we should optimize towards interpreting as nearly everything gets JIT compiled.

Reply:
sounds good, no further comments then